pkg/coordinator/coordinator.go

```diff
@@ -325,9 +401,15 @@ func (rc *reportCoordinator) run() {
 		case <-timer.C:
 			startTime := time.Now()
 
-			rc.checkLogs()
+			// limit the check process to 5 times the cadence as an upper bound
+			ctx, cancel := context.WithTimeout(context.Background(), 5*cadence)
+			if err := rc.checkLogs(ctx); err != nil {
```

Comment:
Not sure of this behaviour. Previously say there was an overload of logs and it took 10 seconds to fetch logs from DB we would have still done that and cleared the cache slowly. Now we will continuously error out. 

What would be the case here for erroring out after 5s?

Reply:
I like the idea of an upper bound on the time the database can take. Database queries already have a timeout anyway so the only reason for running this function to completion of the query is to just rely on that timeout. We can use that method instead.